src/mail_room.py

```diff
@@ -0,0 +1,169 @@
+#  _*_ coding:utf-8 _*_
+"""A simple python program to keep track of names and donations."""
+import sys
+import io
+
+
+GIFTER_DICT = {}
+
+prompts = ["Would you like to 'Send a Thank you' or 'Create a Report'?", "Enter the donator name or press 'L' to see a list of previous names", "How much is the donation?"]
+email_text = "Dear {},\nThank you so much for you donation of ${}. You are a good person.\n"
+
+
+def main_controller():
```

Comment:
There seem to be a few bugs in your execution:

1. I'm only able to generate a list of previous names once per attempt at sending a thank you. If I input 'L' a second time, it's used for the name of a donor.

2. At that same point, if I've input 'L' once to see the previous names and then I press 'q' to get out of the prompt, 'q' gets used for the donor name instead of quitting me out.